src/display/annotation_layer.js

```diff
@@ -3221,8 +3212,7 @@ class AnnotationLayer {
       }
       const isPopupAnnotation = data.annotationType === AnnotationType.POPUP;
       if (!isPopupAnnotation) {
-        const { width, height } = getRectDims(data.rect);
-        if (width <= 0 || height <= 0) {
```

Comment:
This code also handles negative widths/heights, which I assume is to handle cases where the rectangle data is incorrect such that a subtraction would lead to a negative number. However, I'm not sure if we have actually seen such cases or if this code is just defensive. In any case, is this something the new code should also handle or is it not really a concern?

Reply:
- Usually it shouldn't be possible for the width/height to be negative, since we normalize the /Rect-entry; see https://github.com/mozilla/pdf.js/blob/b48717a99e32377831c6098bd833b592756e0335/src/core/annotation.js#L954-L956

 - However, I just noticed that we also have the following code where that assumption may not hold
https://github.com/mozilla/pdf.js/blob/b48717a99e32377831c6098bd833b592756e0335/src/core/annotation.js#L3745-L3749
  To me that "best" solution would be to update that code to re-normalize the rect-data, such that we can always rely upon the width/height being reasonable and non-negative. @timvandermeij Do you agree with this solution, or do you want me to re-introduce the old `<=` checks instead?

Reply:
Good point. I think the current code is OK then given that no reference tests failed as well.